papiea-engine/src/intentful_core/sfs_compiler.ts

```
@@ -16,7 +17,7 @@ export class SFSCompiler {
             sfs_parser(signature)
         } catch(e) {
             throw new ValidationError([
-                new Error(`SFS: '${signature}' parsing on kind: ${kind_name} failed with error: ${e.message}`)
+                new PapieaException(`SFS parsing on kind failed with error: ${e.message}`, { kind_name: kind_name, additional_info: { "signature": signature }})
```

Please add kind coordinates {provider/version/kind}. Perhaps rename `signature` to SFS, to be consistent with the message. 

Also notice that the stacktrace is swallowed, lets find a way to always aggregate stacktraces, never drop them.

---

@Emulebest 
We should discuss how to retain the stack trace for the errors. I think it'll involve changes in our error implementations.

---

The error stacktraces are stored in `error.stack`. One possible solution is keeping a tree-like structure of error references so that in the `PapieaErrorResponseImpl` we can traverse, aggregate and log the stacktraces. This could be achieve by providing optional 3rd or 2nd argument where error nesting is due or by traversing the array of nested errors in case of `ValidationError`

---

I don't see the stack trace getting swallowed in cases of nested errors. I tried to print the stack trace for the above statement and got the following in the logs:
```
papiea-engine    | Error: ["SFS parsing on kind failed with error: Parse error at line 1, column 6:\nwrong, wrong2\n     ^\nExpected:\n\".\"\n\n"]
papiea-engine    |     at Function.try_parse_sfs (/code/papiea-engine/src/intentful_core/sfs_compiler.ts:21:19)
papiea-engine    |     at /code/papiea-engine/src/validator/index.ts:194:70
papiea-engine    |     at Array.forEach (<anonymous>)
papiea-engine    |     at ValidatorImpl.validate_sfs (/code/papiea-engine/src/validator/index.ts:194:43)
papiea-engine    |     at Provider_API_Impl.register_provider (/code/papiea-engine/src/provider/provider_api_impl.ts:48:24)
papiea-engine    |     at processTicksAndRejections (internal/process/task_queues.js:97:5)
papiea-engine    |     at /code/papiea-engine/src/provider/provider_routes.ts:11:24
```
Maybe we can add some field in the `PapieaErrorResponseImpl` to ensure we get a valid stack trace for all the error types